ca_tools/scripts/pos_controller_pid.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+
+import rospy
+import math
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Twist, Pose2D
+from tf.transformations import euler_from_quaternion
+
+
+class DrawSquare:
+    """
+    Class that contains all the functions of the program.
+    """
+    def __init__(self):
+        """
+        Initialization of the variables that are going to be used in the class.
+        """
+
+        #ROS variables
+
+        self.twist_publisher = rospy.Publisher('/create1/cmd_vel', Twist, queue_size=10)
+        self.odom_subscriber = rospy.Subscriber("/create1/gts", Odometry, self.callback)
+
+        # Time variables
+
+        self._now = 0.0
+        self._time_change = 0.0
+        self._last_time = 0.0
+
+        # Angle variables
+
+        self._current_angle = 0.0
+        self._set_angle = 0.0
+        self._input_angle = 0.0
+        self._error_angle = 0.0
+        self._err_sum_angle = 0.0
+        self._output_angle = 0.0
+        self._control_error_angle = 0.01
+
+        # Lenght variables
+
+        self._set_length = Pose2D()
```

Comment:
Why is the length a pose? That doesn't make sense. Probably the variable needs to be named differently